speechbrain/lobes/models/wav2vec.py

```diff
@@ -0,0 +1,610 @@
+import math
+import logging
+import torch
+import torch.nn.functional as F
+import torch.nn as nn
+import random
+import numpy as np
+from contextlib import contextmanager
+import time
+
+from speechbrain.lobes.models.transformer.Transformer import (
+    TransformerEncoder,
+    PositionalEncoding,
+)
+from speechbrain.utils.data_utils import batch_pad_right
+from torch.utils.data._utils.collate import default_convert
+from speechbrain.utils.data_utils import mod_default_collate
+from speechbrain.utils.data_utils import recursive_to
+from torch.utils.data._utils.pin_memory import (
```

Comment:
Why do we need this ?

Reply:
need for W2VPaddedBatch answered elsewhere